Only show bound-with note in holdings when 590$c is present

jkeck opened the pull request:
Closes #1092 

Note 6d7eadc adds a Base MARC Field parsing class that handles most,if not all of the universal parsing code.

1ac3d40 build on top of that class to allow us to easily pull out the bound-with note.

My hope is that we can use (and enhance) this new MARC field parsing class for the upcoming MARC issues when possible, instead of trying to coerce the existing parsing code.

## 3942974 (has $c)
<img width="391" alt="3942974-holdings" src="https://cloud.githubusercontent.com/assets/96776/12740261/64796630-c926-11e5-8c29-d73dff3ae794.png">

## 351225 Before (no $c)
<img width="385" alt="351225-before" src="https://cloud.githubusercontent.com/assets/96776/12740263/647b015c-c926-11e5-90e1-ddadceb5b852.png">

## 351225 After (no $c)
<img width="371" alt="351225-after" src="https://cloud.githubusercontent.com/assets/96776/12740262/647a7c8c-c926-11e5-805e-0761ad171c64.png">

ndushay commented on app/models/marc_field.rb:
```diff
@@ -0,0 +1,138 @@
+    end
+  end
+
+  def tag_indexes
```
skimming this class, everything makes sense to me except this method.  Perhaps a method comment would be useful?

jkeck replied:
Will add a comment.

ndushay commented:
this looks good to me except for me not understanding the function of the tag_indexes method.  Oh ... and should any of the model methods be protected or anything?

jkeck commented:
@ndushay I don't understand what you mean by "should any of the model methods be protected or anything?".

The only public methods in `MarcField` are `inspect`, `label`, 'values`, and `to_partial_path`.  Everything else is private.

ndushay commented:
d'oh - missed the private.  Yay me.  So just confused by the one method - perhaps not so crucial.

jkeck commented:
:+1: Added a comment, rebased, and pushed.  Let me know if the comment doesn't makes sense or if you see some other things I should add.

Thanks for the :eyes:

ndushay commented:
:+1:   when it's ready, go for it.
